data-prepper-plugins/otel-metrics-raw-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/otelmetrics/OTelMetricsRawProcessor.java

```diff
@@ -36,34 +37,41 @@
+
     @DataPrepperPluginConstructor
     public OTelMetricsRawProcessor(PluginSetting pluginSetting, final OtelMetricsRawProcessorConfig otelMetricsRawProcessorConfig) {
         super(pluginSetting);
         this.otelMetricsRawProcessorConfig = otelMetricsRawProcessorConfig;
+        recordsDroppedMetricsRawCounter = pluginMetrics.counter(RECORDS_DROPPED_METRICS_RAW);
     }
 
     @Override
     public Collection<Record<? extends Metric>> doExecute(Collection<Record<ExportMetricsServiceRequest>> records) {
         Collection<Record<? extends Metric>> recordsOut = new ArrayList<>();
         for (Record<ExportMetricsServiceRequest> ets : records) {
-            for (ResourceMetrics rs : ets.getData().getResourceMetricsList()) {
-                final String schemaUrl = rs.getSchemaUrl();
-                final Map<String, Object> resourceAttributes = OTelProtoCodec.getResourceAttributes(rs.getResource());
-                final String serviceName = OTelProtoCodec.getServiceName(rs.getResource()).orElse(null);
+            try {
+                for (ResourceMetrics rs : ets.getData().getResourceMetricsList()) {
+                    final String schemaUrl = rs.getSchemaUrl();
+                    final Map<String, Object> resourceAttributes = OTelProtoCodec.getResourceAttributes(rs.getResource());
+                    final String serviceName = OTelProtoCodec.getServiceName(rs.getResource()).orElse(null);
 
-                for (InstrumentationLibraryMetrics is : rs.getInstrumentationLibraryMetricsList()) {
-                    final Map<String, Object> ils = OTelProtoCodec.getInstrumentationLibraryAttributes(is.getInstrumentationLibrary());
-                    recordsOut.addAll(processMetricsList(is.getMetricsList(), serviceName, ils, resourceAttributes, schemaUrl));
-                }
+                    for (InstrumentationLibraryMetrics is : rs.getInstrumentationLibraryMetricsList()) {
+                        final Map<String, Object> ils = OTelProtoCodec.getInstrumentationLibraryAttributes(is.getInstrumentationLibrary());
+                        recordsOut.addAll(processMetricsList(is.getMetricsList(), serviceName, ils, resourceAttributes, schemaUrl));
+                    }
 
-                for (ScopeMetrics sm : rs.getScopeMetricsList()) {
-                    final Map<String, Object> ils = OTelProtoCodec.getInstrumentationScopeAttributes(sm.getScope());
-                    recordsOut.addAll(processMetricsList(sm.getMetricsList(), serviceName, ils, resourceAttributes, schemaUrl));
+                    for (ScopeMetrics sm : rs.getScopeMetricsList()) {
+                        final Map<String, Object> ils = OTelProtoCodec.getInstrumentationScopeAttributes(sm.getScope());
+                        recordsOut.addAll(processMetricsList(sm.getMetricsList(), serviceName, ils, resourceAttributes, schemaUrl));
+                    }
                 }
-
+            } catch (Exception e) {
```

Comment:
I tend to think we should put this exception handling inside the loop in `processMetricsList`.

Say we get a metrics export request with 50 metrics and only one is invalid. The current try-catch will fail them all. I think we can probably isolate these exceptions to the individual metrics and thus allow the metrics which have the correct pre-condition to process correctly.

```
      for (io.opentelemetry.proto.metrics.v1.Metric metric : metricsList) {
        try {
            if (metric.hasGauge()) {
                recordsOut.addAll(mapGauge(metric, serviceName, ils, resourceAttributes, schemaUrl));
            } else if (metric.hasSum()) {
                recordsOut.addAll(mapSum(metric, serviceName, ils, resourceAttributes, schemaUrl));
            } else if (metric.hasSummary()) {
                recordsOut.addAll(mapSummary(metric, serviceName, ils, resourceAttributes, schemaUrl));
            } else if (metric.hasHistogram()) {
                recordsOut.addAll(mapHistogram(metric, serviceName, ils, resourceAttributes, schemaUrl));
            } else if (metric.hasExponentialHistogram()) {
                recordsOut.addAll(mapExponentialHistogram(metric, serviceName, ils, resourceAttributes, schemaUrl));
            }
        } catch(Exception ex) {
        ...
        }
      }

```

Reply:
Makes sense. I thought each record in doExecute() corresponds to one metric but it looks like it can result in multiple metrics. I will move the try-catch to `processMetricsList`